### ISSUE TITLE
unexpected behavior when click on box

### ISSUE THREAD

**Issue by stefanocudini** [ScanEx/Leaflet-IconLayers#5]
I think this is strange behavior for a user.
It would be much better if the selected layer remains in the same position

**Comment by aparshin** [comment]
I agree that it is a little bit strange to rotate layers in control after selection. The advantage of such behavior is that you can switch two (and even three) last layers by single mouse click.

We are going to implement other behaviors and add corresponding option in control.

**Comment by stefanocudini** [comment]
What seem to me is the rotation is random and unpredictable :( ... also in this demo:
http://scanex.github.io/Leaflet-IconLayers/examples/

Best way is maintain same position and adding class for selection.

Besides these it's an great plugin!

**Comment by aparshin** [comment]
> What seem to me is the rotation is random and unpredictable :( ... also in this demo:
> http://scanex.github.io/Leaflet-IconLayers/examples/

Well, uh... Looks like something is really wrong here :) I'll check it, thanks!

> Best way is maintain same position and adding class for selection.

The main idea of current algorithm is to show "best next layer" in bottom left corner of the widget (it allows users to select that layer by single click). Such behavior changes layer positions. As I already said, we are going to implement multiple variants of layer selection behavior...

> Besides these it's an great plugin!

Thank you!

**Comment by EmiyaGm** [comment]
I've tried to solve this problem, so you can look at my code
[https://github.com/EmiyaGm/Leaflet-IconLayers](url)

**Comment by jinixx** [comment]
Thanks for the great plugin guys.

Our users are however, complaining about the same, will there be an update to this?

**Comment by marie-becker** [comment]
Do you still plan on updating the plugin? Our clients unfortunately complained about it being confusing
